sdk/maps/azure-maps-search/src/test/java/com/azure/maps/search/MapsSearchAsyncClientTest.java

```diff
@@ -0,0 +1,722 @@
+import com.azure.maps.search.models.SearchPointOfInterestOptions;
+import com.azure.maps.search.models.SearchStructuredAddressOptions;
+import com.azure.maps.search.models.StructuredAddress;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.File;
+import java.io.IOException;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import reactor.test.StepVerifier;
+
+public class MapsSearchAsyncClientTest extends MapsSearchClientTestBase {
+    private MapsSearchAsyncClient client;
+    private static final String DISPLAY_NAME_WITH_ARGUMENTS = "{displayName} with [{arguments}]";
+
+    @BeforeAll
+    public static void beforeAll() {
+        StepVerifier.setDefaultTimeout(Duration.ofSeconds(30));
+    }
+
+    @AfterAll
+    public static void afterAll() {
+        StepVerifier.resetDefaultTimeout();
+    }
+
+    private MapsSearchAsyncClient getMapsSearchAsyncClient(HttpClient httpClient,
+        MapsSearchServiceVersion serviceVersion) {
+        return getMapsSearchAsyncClientBuilder(httpClient, serviceVersion).buildAsyncClient();
+    }
+
+    // Test async get polygons
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.maps.search.TestUtils#getTestParameters")
+    public void testAsyncGetMultiPolygons(HttpClient httpClient, MapsSearchServiceVersion serviceVersion) {
+        client = getMapsSearchAsyncClient(httpClient, serviceVersion);
+        List<String> geometryIds = Arrays.asList("8bceafe8-3d98-4445-b29b-fd81d3e9adf5", "00005858-5800-1200-0000-0000773694ca");
+        StepVerifier.create(client.getPolygons(geometryIds)).assertNext(actualResults -> 
+        {
+            try {
+                validateGetPolygons(TestUtils.getMultiPolygonsResults(), actualResults);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }).verifyComplete();
+    }
+
+    // Test async get polygons with response
+
+    // Case 1: 200
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.maps.search.TestUtils#getTestParameters")
+    public void testAsyncGetPolygonsWithResponse(HttpClient httpClient, MapsSearchServiceVersion serviceVersion) {
+        client = getMapsSearchAsyncClient(httpClient, serviceVersion);
+        List<String> geometryIds = Arrays.asList("8bceafe8-3d98-4445-b29b-fd81d3e9adf5", "00005858-5800-1200-0000-0000773694ca");
+        StepVerifier.create(client.getPolygonsWithResponse(geometryIds, Context.NONE))
+                .assertNext(response ->
+                {
+                    try {
+                        validateGetPolygonsWithResponse(TestUtils.getMultiPolygonsResults(), 200, response);
+                    } catch (IOException e) {
+                        e.printStackTrace();
```

Comment:
same comments about the exception catching. better just let it blow and see if it successfully fails the test